llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -1868,7 +1868,7 @@ bool NVPTXDAGToDAGISel::tryStoreParam(SDNode *N) {
     case 1: {
       MVT::SimpleValueType MemTy = Mem->getMemoryVT().getSimpleVT().SimpleTy;
       SDValue Imm = Ops[0];
-      if (MemTy != MVT::f16 && MemTy != MVT::v2f16 &&
+      if (MemTy != MVT::f16 && MemTy != MVT::bf16 &&
```

Comment:
Was `v2f16` removed intentionally?

We do have some corner cases where we have vectors of vectors, specifically for 16-bit FP types.

Reply:
I removed it in response to Alex's comment under the belief that it was impossible to create a node of type ConstantSDNode or ConstantFPSDNode of vector type. If that assumption is incorrect I'm happy to add it back in. However, a test that fails without it and succeeds with it would be required. Can you provide the corner case so I can add a test?

Reply:
OK. It may indeed be the case. Sorry about the noise, I've looked at the last version of the diff and github unhelpfully didn't show the conversation on exactly this point. I can see it on the Conversation tab now.